### ISSUE TITLE
Zone names might end with a "."

### ISSUE THREAD

**Issue by vquicksilver** [joeig/go-powerdns#34]
Hi,

I was trying to setup today 2 different cert-manager webhooks today which appear to have in common this same library (one uses v2 and the other one v3). After fighting with both for a while we realized that zonenames are being trimmed in what it seems an incorrect way.

For example we have "subdomain-test.main.com." as a zone in our powerdns service, and somehow the final "." is being trimmed when the webhooks call pdns.Zones.Get(ctx, "zonename").

Is there any reason why we are trimming the "." at the end?.

What is even funnier is that if I append 2 "." to my zonename in my webhook configuration, I can make the final "." pass the trim function, so one of the dots don't get removed.

Thanks.

**Comment by joeig** [comment]
Hi, which version of PowerDNS are you using?

**Comment by Habbie** [comment]
We (I work with @vquicksilver) are using [ZoneControl](https://www.open-xchange.com/products/ox-powerdns-authoritative/) (not open source), which, besides a control panel, is also an API proxy that supports multiple backends (so we're very happy that go-powerdns has Vhost support!).

It turns out (we found out today) that ZoneControl is stricter about paths than PowerDNS is.

Anyway, the short version is that the trailing dot is expected in the API-call - PowerDNS just apparently also accepts a path without one. Ideally (and I notice we did not document that very explicitly) paths are acquired indirectly:

```
$ curl  -s -H 'X-API-Key: apikey' http://127.0.0.1:8081/api/v1/servers/localhost/zones?zone=7bits.nl. | jq
[
  {
    "account": "",
    "dnssec": true,
    "edited_serial": 134,
    "id": "7bits.nl.",
    "kind": "Slave",
    "last_check": 1644862311,
    "masters": [
      "89.188.0.34"
    ],
    "name": "7bits.nl.",
    "notified_serial": 132,
    "serial": 134,
    "url": "/api/v1/servers/localhost/zones/7bits.nl."
  }
]
```

so that weird characters can be encoded for you by PowerDNS.

In summary, adding the trailing dot will make go-powerdns work with other products that claim PowerDNS API-compatibility, and will still work correctly with the normal PowerDNS server. Longer term it might make sense to do the indirection via `../zones?zone=..` for extra robustness.

**Comment by joeig** [comment]
Thanks for your feedback. The implementation will be part of v3.0.3.
